test/sslapitest.c

```diff
@@ -2027,6 +2239,97 @@ static int test_tlsext_status_type(void)
 
     return testresult;
 }
+
+static int test_tlsext_status_type_multi(void)
+{
+    SSL_CTX *cctx = NULL, *sctx = NULL;
+    SSL *clientssl = NULL, *serverssl = NULL;
+    int testresult = 0;
+    char *leaf_chain = test_mk_file_path(certsdir, "leaf-chain.pem");
+    char *skey = test_mk_file_path(certsdir, "leaf.key");
+    char *leaf = test_mk_file_path(certsdir, "leaf.pem");
+    OSSL_LIB_CTX *tmpctx;
+
+    tmpctx = OSSL_LIB_CTX_set0_default(libctx);
+
+    if (!create_ssl_ctx_pair(libctx, TLS_server_method(), TLS_client_method(),
+                             TLS1_VERSION, 0,
+                             &sctx, &cctx, leaf, skey))
+        return 0;
+
+    if (SSL_CTX_use_certificate_chain_file(sctx, leaf_chain) <= 0)
+        return 0;
+
+    if (SSL_CTX_get_tlsext_status_type(cctx) != -1)
+        goto end;
+
+    /* First just do various checks getting and setting tlsext_status_type */
+
+    clientssl = SSL_new(cctx);
+    if (!TEST_ptr(clientssl))
+        goto end;
+    if (!TEST_int_eq(SSL_get_tlsext_status_type(clientssl), -1)
+            || !TEST_true(SSL_set_tlsext_status_type(clientssl,
+                                                     TLSEXT_STATUSTYPE_ocsp))
+            || !TEST_int_eq(SSL_get_tlsext_status_type(clientssl),
+                            TLSEXT_STATUSTYPE_ocsp))
+        goto end;
+
+    SSL_free(clientssl);
+    clientssl = NULL;
+
+    if (!SSL_CTX_set_tlsext_status_type(cctx, TLSEXT_STATUSTYPE_ocsp)
+        || SSL_CTX_get_tlsext_status_type(cctx) != TLSEXT_STATUSTYPE_ocsp)
+        goto end;
+
+    /*
```

Comment:
It would be more intuitive to do the test with an OCSP response with just the server cert first,
and then the test with multiple responses.